### ISSUE TITLE
god_function flags __init__ methods with many parameters

### ISSUE THREAD

**Issue by rsionnach** [rsionnach/sloppylint#9]
## Issue
The `god_function` pattern flags `__init__` methods that have more than 5 parameters, but complex class initialization often legitimately requires many parameters.

## Example
```python
class DatabaseConnection:
    def __init__(self, host, port, user, password, database, timeout):
        # FLAGGED for 6 parameters - but this is valid initialization!
        self.host = host
        self.port = port
        # ...
```

## Expected Behavior
`__init__` methods should have a higher parameter threshold or be exempt from the parameter count check, as they often need to accept many configuration values.

## Suggested Fix
- Increase MAX_PARAMS for `__init__` methods (e.g., to 8-10)
- Or exempt `__init__` and `__new__` from parameter count checks entirely
- Consider also exempting methods decorated with `@dataclass` style patterns

**Comment by rsionnach** [comment]
This was already addressed in a previous release. The `god_function` pattern uses `MAX_PARAMS_INIT = 10` for `__init__` and `__new__` methods, allowing up to 10 parameters for constructors vs 5 for regular functions.

```python
MAX_PARAMS = 5
MAX_PARAMS_INIT = 10  # Higher threshold for __init__ methods
```

Closing as fixed.
